knr1050Sup/knr1050.db

```diff
@@ -26,127 +45,316 @@ record(bi, "$(P)DISABLE") {
-	field(ZNAM, "")
-	field(ONAM, "Remote")
+	field(INP, "@knr1050.proto get_remote_mode($(P)MODE) $(PORT)")
 }
 
-record(bo, "$(P)LOCAL_MODE:SP") {
-	field(DESC, "Sets the device into local mode")
-	field(DTYP, "stream")
+record(bo, "$(P)MODE") {
+	field(DESC, "Current remote/local mode state")
 	field(SCAN, "Passive")
 	
-	field(OUT, "@knr1050.proto set_local_mode() $(PORT)")
-	field(ZNAM, "")
-	field(ONAM, "Local")
+	field(ZNAM, "LOCAL")
+	field(ONAM, "REMOTE")
+
+	info(INTEREST, "HIGH")
 }
 
-record(bi, "$(P)GET_REMOTE_MODE") {
-	field(DESC, "Gets the current remote state")
+record(bo, "$(P)MODE:SP") {
+	field(DESC, "Sets the device mode")
 	field(DTYP, "stream")
-	field(SCAN, "1 second")
 	
-	field(INP, "@knr1050.proto get_remote_mode($(P)REMOTE_MODE) $(PORT)")
+	field(OUT, "@knr1050.proto set_mode $(PORT)")
+	
+	field(ZNAM, "LOCAL")
+	field(ONAM, "REMOTE")
 }
 
-record(bo, "$(P)REMOTE_MODE") {
-	field(DESC, "Current remote/local mode state")
-	field(SCAN, "Passive")
+alias("$(P)MODE", "$(P)MODE:SP:RBV")
+
+###########################################################
+# 3. PUMP CONTROL
+###########################################################
+# 
+#  The user can either begin a continuous pump (START:SP),
+#  or, pump for a set time/volume (TIMED:SP). For a timed run
+#  a desired run time or volume must be specified (see Sec.4).
+#
+#  If the device is in local mode, or the gradients
+#  are not correctly set, then the pump start/timed set point
+#  is disabled.
+#
+record(sseq, "$(P)STOP:SP")
+{
+	field(DESC, "Stop pump")
+	
+	field(LNK1, "$(P)_PUMP:STOP:SP CA")
+	field(DO1, "1")
+    field(WAIT1, "Wait")
+	
+	field(LNK2, "$(P)TIME:STATUS PP")
+	field(DO2, "0")
 	
-	field(ZNAM, "Local")
-	field(ONAM, "Remote")
+	field(DISV, "0")
+	field(SDIS, "$(P)DISABLE")
+	info(INTEREST, "HIGH")
+}
+
+alias("$(P)STOP:SP", "$(P)STOP")
 
+record(sseq, "$(P)TIMED:SP")
+{
+	field(DESC, "Start timed pump for set time or volume")
+	
+	field(LNK1, "$(P)_PUMP:START:SP CA")
+	field(DO1, "1")
+    field(WAIT1, "Wait")
+	
+	field(LNK2, "$(P)TIME:START PP")
+	field(DO2, "1")
+	
+	field(LNK3, "$(P)TIME:STATUS PP")
+	field(DO3, "1")
+	
+	field(LNK4, "$(P)TIME PP")
+	field(DO4, "0")
+	
+	field(DISV, "0")
+	field(SDIS, "$(P)DISABLE")
 	info(INTEREST, "HIGH")
 }
 
-###########################################################
-# Control the pump to either begin or stop a ramp
-###########################################################
+alias("$(P)TIMED:SP", "$(P)TIMED")
 
-# ///
-# /// The pump start set point calls both a ramp command as
-# /// well as a forward linked stop mode command. The stop
-# /// mode keeps the last values rather than stopping the
-# /// pump. If the device is in local mode, or the gradients
-# /// are not correctly set, then the pump start set point is
-# /// disabled.
-# ///
-record(bo, "$(P)PUMP:STOP:SP") {
-	field(DESC, "Stops the pump")
-	field(DTYP, "stream")
-	field(SCAN, "Passive")
+record(sseq, "$(P)START:SP")
```

Comment:
The reason for this is the included `WAIT`. We use the sseq because it will process the link and ensure it has completed before we continue to performing any other steps (starting the counter etc).